haystack/modeling/model/triadaptive_model.py

```diff
@@ -310,9 +310,9 @@ def forward_lm(self, **kwargs):
             # Current batch consists of only tables
             if all(table_mask):
                 pooled_output2, _ = self.language_model3(
-                    passage_input_ids=kwargs["passage_input_ids"],
-                    passage_segment_ids=kwargs["table_segment_ids"],
-                    passage_attention_mask=kwargs["passage_attention_mask"],
+                    input_ids=kwargs["passage_input_ids"],
+                    segment_ids=kwargs["table_segment_ids"],
+                    attention_mask=kwargs["passage_attention_mask"],
                     output_hidden_states=False,
                     output_attentions=False,
                 )
```

Comment:
Let's make the input two-dimensional here, similar to the case where we only have text Documents.
```suggestion
            if all(table_mask):
                # Make input two-dimensional
                max_seq_len = kwargs["table_input_ids"].shape[-1]
                input_ids = kwargs["table_input_ids"].view(-1, max_seq_len)
                attention_mask = kwargs["table_attention_mask"].view(-1, max_seq_len)
                segment_ids = kwargs["table_segment_ids"].view(-1, max_seq_len)

                pooled_output2, _ = self.language_model3(
                    input_ids=input_ids,
                    segment_ids=segment_ids,
                    attention_mask=attention_mask,
                    output_hidden_states=False,
                    output_attentions=False,
                )
```